sopht_mpi/utils/mpi_io.py

```diff
@@ -0,0 +1,772 @@
+"""Module for Input/Output via HD5 format."""
+import h5py
+import numpy as np
+from mpi4py import MPI
+from elastica.rod.cosserat_rod import CosseratRod
+
+
+class MPIIO:
+    r"""MPI-supported IO class for field save and load.
```

Comment:
A question here, is it possible derive this class from IO in `sopht` to maybe reduce some code repeat and put common interfaces/methods at one place?

Reply:
I wanted to do that initially, but as it seems (at least for now, subject to possible refactoring in the future), not only does the `save()` method need to be overloaded (like we have in `cosseratIO` in `sopht`), other methods like `define_eulerian_grid()` and `add_as_*_fields()` also needs some minor additions to include some information that pertains to the mpi local domain bounds and ghost cells etc.

One thing we could perhaps think about in the future is perhaps making the "metadata structure" that we currently have (see the low-budget illustration I made in the docstring) as an abstraction that is passed into `IO` so that we can write a `save()` method that traverses through the abstracted metadata structure and save fields accordingly. It is not clear to me yet how we will do it, but does sound plausible to me. If that is done, then perhaps we can derive `MPIIO` from `IO`, both of which takes in the metadata structure and call their `save()` function accordingly. We could look into this if you are interested, but I would probably put it as a low-priority issue until I get the solver assembled and tested on clusters. Is that fine @bhosale2 ?

Reply:
@bhosale2 Oh another thing we can think about is to perhaps separate eulerian and lagrangian field IO, with both deriving from a base IO class with abstract methods like `add_fields()`, `save()`, `load()` and `generate_xdmf()`. In any case, I think we have a few ways to go about this and it might be a good idea if we can discuss this in person sometime. Thanks!

Reply:
Ok please create an issue for this and update this thread with the issue link. We will then move ahead with this PR.

Reply:
Okay I've opened issue #146 to take care of the stuff discussed here in the future. Thanks!